app/scripts/metamask-controller.js

```diff
@@ -2677,8 +2735,10 @@ export default class MetamaskController extends EventEmitter {
       addTransaction: this.addTransaction.bind(this),
       addTransactionAndWaitForPublish:
         this.addTransactionAndWaitForPublish.bind(this),
-      createTransactionEventFragment:
-        txController.createTransactionEventFragment.bind(txController),
+      createTransactionEventFragmentWithTxId:
```

Comment:
I appreciate this might be the more accurate name, but for the sake of avoiding all the UI changes in this PR and general consistency, could we leave the property name as `createTransactionEventFragment` but bind to our new `createTransactionEventFragmentWithTxId` method?

Reply:
Rolled back in this commit https://github.com/MetaMask/metamask-extension/pull/21330/commits/22e50c9635789db378ed0c925a3e1eebefc28c8b